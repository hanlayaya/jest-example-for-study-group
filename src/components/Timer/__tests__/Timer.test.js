/* eslint-disable import/first */
jest.mock('lodash.debounce', () => {
  return func => func;
});

import React from 'react';
import { mount } from 'enzyme';
import MockDate from 'mockdate';
import styles from '../Timer.scss';

describe('<Timer />', function () {
  let Timer;

  beforeEach(() => {
    jest.resetAllMocks();
    jest.useFakeTimers();
    // Mocked Date
    MockDate.set(1501951314962);
    Timer = require('../Timer').default;

    this.params = {
      load: jest.fn(),
    };

    this.makeSubject = () => {
      const { load } = this.params;

      return mount(
        <Timer
          load={load}
        />
      );
    };
  });

  describe('when it has no any data', () => {
    beforeEach(() => {
      this.subject = this.makeSubject();
    });

    it('should be same as snapshot', () => {
      expect(this.subject.html()).toMatchSnapshot();
    });
  });

  describe('simulate click behavior', () => {

    beforeEach(() => {
      this.subject = this.makeSubject();
    });

    it('should call load function', () => {
      MockDate.set(1501954373526);
      this.subject.find(`.${styles.refresh}`).simulate('click');

      jest.runAllTimers();

      expect(this.params.load).toBeCalled();
      expect(this.subject.state().now).toMatchSnapshot();
    });
  });
});
