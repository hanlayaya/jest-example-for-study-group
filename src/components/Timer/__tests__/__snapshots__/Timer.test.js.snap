// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Timer /> simulate click behavior should call load function 1`] = `"Sun Aug 06 2017 01:32:53 GMT+0800 (CST)"`;

exports[`<Timer /> when it has no any data should be same as snapshot 1`] = `"<div><span><!-- react-text: 3 -->資料最後更新時間：<!-- /react-text --><!-- react-text: 4 -->Sun Aug 06 2017 00:41:54 GMT+0800 (CST)<!-- /react-text --></span><a class=\\"refresh\\">更新</a></div>"`;
